bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/PreviewServlet.java

```diff
@@ -93,10 +92,7 @@ private Page getRefererPage(SlingHttpServletRequest request) {
 
         int idx = referer.lastIndexOf(PAGE_EDITOR_PATH);
         String pagePath = referer.substring(idx + PAGE_EDITOR_PATH.length());
-        pagePath = pagePath.replaceAll(".html", ""); // get rid of .html extension
-
-        ResourceResolver resourceResolver = request.getResourceResolver();
-        Resource pageResource = resourceResolver.getResource(pagePath);
-        return pageResource != null ? pageResource.adaptTo(Page.class) : null;
+        Resource pageResource = request.getResourceResolver().resolve(pagePath);
+        return pageResource.adaptTo(Page.class);
```

Comment:
Why is the null check removed here? There might be some issue with the request parameter and pagePath can be empty or wrong.

Reply:
ResourceResolver.resolve() differs from ResourceResolver.getResource() by returning NonExistentResource instead of null and then adaptTo(Page.class) will return null.